### PR TITLE
new resource "azurerm_spring_cloud_app_redis_association"

### PR DESCRIPTION
**Pull request by njuCZ**
spring cloud app could bind to different service, such as redis, cosmosDB, mysql

this is the first PR to bind redis 

image

### CODE REVIEW

File: azurerm/internal/services/springcloud/spring_cloud_app_redis_binding_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r SpringCloudAppRedisBindingResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_app_redis_binding" "test" {
+  name                = "acctestscarb-%d"
```
Comment (katbyte) [inline azurerm/internal/services/springcloud/spring_cloud_app_redis_binding_resource_test.go:104]: are capitals allowed here? if so we should use some

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @njuCZ - overall this looks good and i just have some minor comments to address before merging

File: website/docs/r/spring_cloud_app_redis_binding.html.markdown [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+subcategory: "Spring Cloud"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_spring_cloud_app_redis_binding"
+description: |-
+  Manages an Azure Spring Cloud Application Redis Binding instance.
+---
+
+# azurerm_spring_cloud_app_redis_binding
+
+Manages an Azure Spring Cloud Application Redis Binding instance.
```
[the remark this answers is not in the document]
Comment (njuCZ) [reply website/docs/r/spring_cloud_app_redis_binding.html.markdown:11]: done

File: azurerm/internal/services/springcloud/registration.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ func (r Registration) SupportedResources() map[string]*schema.Resource {
 	return map[string]*schema.Resource{
 		"azurerm_spring_cloud_active_deployment": resourceSpringCloudActiveDeployment(),
 		"azurerm_spring_cloud_app":               resourceSpringCloudApp(),
+		"azurerm_spring_cloud_app_redis_binding": resourceSpringCloudAppRedisBinding(),
```
[the remark this answers is not in the document]
Comment (njuCZ) [reply azurerm/internal/services/springcloud/registration.go:34]: thanks for the suggestions

File: azurerm/internal/services/springcloud/spring_cloud_app_redis_binding_resource_test.go (hunk 1 above)
Comment (njuCZ) [reply azurerm/internal/services/springcloud/spring_cloud_app_redis_binding_resource_test.go:104]: only lower case letters, numbers and hythens are allowed. In the validation func, I have added the rule

**Comment by njuCZ** [comment]
@katbyte I have refactor this resource

image

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - this LGTM now! 👍

**Pull request merged**